de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/genome/GenomeSearchRpcProxy.java

```diff
@@ -0,0 +1,76 @@
+package org.iplantc.de.diskResource.client.presenters.genome;
+
+import org.iplantc.de.client.models.genomes.Genome;
+import org.iplantc.de.client.services.FileEditorServiceFacade;
+import org.iplantc.de.commons.client.ErrorHandler;
+import org.iplantc.de.commons.client.info.ErrorAnnouncementConfig;
+import org.iplantc.de.commons.client.info.IplantAnnouncer;
+import org.iplantc.de.diskResource.client.GenomeSearchView;
+
+import com.google.common.base.Strings;
+import com.google.gwt.user.client.rpc.AsyncCallback;
+
+import com.sencha.gxt.data.client.loader.RpcProxy;
+import com.sencha.gxt.data.shared.loader.FilterConfig;
+import com.sencha.gxt.data.shared.loader.FilterPagingLoadConfig;
+import com.sencha.gxt.data.shared.loader.PagingLoadResult;
+import com.sencha.gxt.data.shared.loader.PagingLoadResultBean;
+
+import java.util.List;
+import javax.inject.Inject;
+
+/**
+ * An RpcProxy class that handles searching genomes when the user has typed a search term in the GenomeSearchView's search field
+ */
+public class GenomeSearchRpcProxy extends RpcProxy<FilterPagingLoadConfig, PagingLoadResult<Genome>> {
+
+    private GenomeSearchView.GenomeSearchViewAppearance appearance;
+    private FileEditorServiceFacade serviceFacade;
+    private IplantAnnouncer announcer;
+    String lastQueryText;
```

Comment:
This could be kept inside the `load` method, since I don't think it's referenced anywhere else, and I don't think we need to keep track of the previous query for anything.

Reply:
Ah true, some copy-pasta problems 😄